### ISSUE TITLE
Modal Dialog focus accessibility issues

### ISSUE THREAD

**Issue by thatchej** [pluralsight/design-system#260]
### Desired behavior

Users are unable to use Tab or Shift+Tab to navigate focus outside of a Modal Dialog

### Today's behavior

Currently it is possible to use Tab or Shift+Tab to move focus outside of the design system modal dialog. Technically speaking, this makes sense and is exactly the behavior you'd expect for a custom dialog/modal. However that so happens to be an accessibility violation

### The Value Add

This would remove the aforementioned accessibility issue across all BCs that use the design system. 

### The Tradeoffs

Must bring in additional 3rd party library or spend a non-trivial amount of time coding a solution. From my research this is a somewhat difficult issue to solve.

**Comment by jaketrent** [comment]
A potential implementation from Andy L. https://pluralsight.slack.com/archives/C70HPSJPP/p1539732119000100?thread_ts=1539730876.000100&cid=C70HPSJPP
